pkg/serverless/metrics/metric.go

```diff
@@ -0,0 +1,138 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package metrics
+
+import (
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/dogstatsd"
+	"github.com/DataDog/datadog-agent/pkg/forwarder"
+	"github.com/DataDog/datadog-agent/pkg/metrics"
+	"github.com/DataDog/datadog-agent/pkg/serializer"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// PlatformObjectRecord contains additional information found in Platform log messages
+type PlatformObjectRecord struct {
+	RequestID string           // uuid; present in LogTypePlatform{Start,End,Report}
+	Version   string           // present in LogTypePlatformStart only
+	Metrics   ReportLogMetrics // present in LogTypePlatformReport only
+}
```

Comment:
I don't think it's needed in this PR but some day it would be nice to have separate types for the different types of object records in the different types of platform logs.

Reply:
Also I agree with Darcy's earlier comment that these should be moved to the logs package. These describe the specification of the logs API, and some of these log lines (Start and End) don't even have any metrics anyway, so these really shouldn't be in the metrics package imo.